src/convertToObject.js

```diff
@@ -3,16 +3,39 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
+const propertySeparator = ';';
+const keySeparator = ':';
+
 function convertToObject(sourceString) {
-  // write your code here
+  const properties = sourceString
+    .split(propertySeparator)
+    // .map(prop => prop.trim().split(keySeparator))
+    // .filter(prop => prop[0] !== '');
+    .reduce((acc, prop) => {
+      const line = prop.trim().split(keySeparator);
+
+      if (line[0] !== '') {
+        acc.push(line);
+      }
+
+      return acc;
+    }, []);
+
+  const styleList = properties.reduce((acc, prop) => {
+    return Object.assign(acc, { [prop[0].trim()]: prop[1].trim() });
+  }, {});
```

Comment:
It can be done in 1 `reduce`. In this case, we will avoid using 1 more loop and increase performance of your code 🙃